### PR TITLE
chore: render diff of changed `seq` items

### PR DESCRIPTION
**Pull request by Lou1415926**
Some logic from #4640 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/diff/formatter.go [hunk 1]
```diff
@@ -32,7 +34,17 @@ func (f *seqItemFormatter) formatMod(node diffNode) string {
...
-	return ""
+	return color.Faint.Sprint("- (changed item)")
+}
+
+func (f *seqItemFormatter) childIndent(curr int) int {
+	/* A seq item diff should look like:
+	   - (item)
+	     ~ Field1: a
+	     + Field2: b
+	   Where "~ Filed1: a" and "+ Field2: b" are its children. The indentation should increase by len("- "), which is 2.
```
Comment (efekarakus) [inline internal/pkg/template/diff/formatter.go]:
```suggestion
	   Where "~ Field1: a" and "+ Field2: b" are its children. The indentation should increase by len("- "), which is 2.
```

File: internal/pkg/template/diff/formatter.go [hunk 2]
```diff
@@ -7,13 +7,15 @@ import (
...
 
+	"github.com/aws/copilot-cli/internal/pkg/term/color"
 	"gopkg.in/yaml.v3"
 )
 
 type formatter interface {
 	formatYAML(*yaml.Node) ([]byte, error)
 	formatMod(node diffNode) string
 	formatPath(node diffNode) string
+	childIndent(curr int) int
```
Comment (efekarakus) [inline internal/pkg/template/diff/formatter.go]: nit:
```suggestion
	nextIndent(curr int) int
```

**Review by efekarakus — APPROVED** [review]
honestly, i love our diff feature

File: internal/pkg/template/diff/formatter.go (hunk 1 above)
Comment (Lou1415926) [reply internal/pkg/template/diff/formatter.go]: i am the king of typo

**Review by huanjani — APPROVED** [review]
🐱 🍓

**Pull request merged**